src/modules/Dashboard/features/RespondentData/RespondentDataReview/RespondentDataReview.tsx

```diff
@@ -34,13 +44,53 @@ export const RespondentDataReview = () => {
   const [isLastVersion, setIsLastVersion] = useState<boolean>(false);
   const [isBannerVisible, setIsBannerVisible] = useState<boolean>(false);
   const [itemIds, setItemIds] = useState<string[]>([]);
+  const [responseDates, setResponseDates] = useState<Date[]>();
+  const [activities, setActivities] = useState<ReviewActivity[]>([]);
   const [isLoading, setIsLoading] = useState(false);
-  const dataTestid = 'respondents-review';
-
   const [activityItemAnswers, setActivityItemAnswers] = useState<
     DecryptedActivityData<EncryptedAnswerSharedProps>['decryptedAnswers'] | null
   >(null);
 
+  const { control } = useForm({
+    defaultValues: {
+      date:
+        selectedDateParam && isValid(new Date(selectedDateParam!))
+          ? new Date(selectedDateParam!)
+          : new Date(),
+    },
+  });
+
+  const date = useWatch({
+    control,
+    name: 'date',
+  });
+
+  const dataTestid = 'respondents-review';
+
+  const { execute: getAppletSubmitDateList } = useAsync(
+    getAppletSubmitDateListApi,
+    (datesApiResult) => {
+      const datesResult = datesApiResult?.data?.result;
+      if (!datesResult) return;
+
+      const submitDates = datesResult.dates.map((date: string) => new Date(date));
+      setResponseDates(submitDates);
+    },
+  );
+
+  // TODO: refactor
+  const { execute: getReviewActivities } = useAsync(getReviewActivitiesApi, (res) => {
+    res?.data?.result && setActivities(res.data.result);
```

Comment:
could you pls add early return?
const activities = res?.data?.result;
if (!activities) return;
...